pkg/core/mpt/node.go

```diff
@@ -27,6 +27,17 @@ type NodeObject struct {
 	Node
 }
 
+// extendedNodeObject extends NodeObject with gc-related info.
+type extendedNodeObject struct {
+	NodeObject
+	RefCount uint32
+}
+
+type nodeWrapper interface {
```

Comment:
I think think interface can be avoided, and better be avoided (as it adds some overhead). You can add `RefCount` to `NodeObject` and deserialize it in `getFromStore` when needed.

Reply:
Fixed.